runtime/parachains/src/session_info.rs

```diff
@@ -196,10 +210,20 @@ impl<T: Config> Pallet<T> {
 			dispute_period,
 		};
 		Sessions::<T>::insert(&new_session_index, &new_session_info);
-		SessionExecutorParams::<T>::insert(
-			&new_session_index,
-			ExecutorParams::from(&EXECUTOR_PARAMS[..]),
-		);
+
+		let new_executor_params = if let Some(executor_params) = <PendingExecutorParams<T>>::get() {
+			<PendingExecutorParams<T>>::put(&None);
+			executor_params
+		} else {
+			let current_session_index = <shared::Pallet<T>>::session_index();
```

Comment:
Is it okay to request the current session index from inside of the `initializer_on_new_session` handler? Wouldn't it return a new session index? If yes, then where can I get the old one? Just `.saturated_sub(1)` from the new one maybe?

Reply:
`initializer_on_new_session` is first called for `shared`, then for everyone else.

For safety you could change the `SessionChangeNotification` to include `prev_session_index` returned by `shared`, I guess.

Reply:
> Is it okay to request the current session index from inside of the initializer_on_new_session handler? Wouldn't it return a new session index? If yes, then where can I get the old one? Just .saturated_sub(1) from the new one maybe?

Yeah, saturated_sub seems fine. The point of having the `initializer_on_new_session` approach is to have a clear view into relative initialization order of pallets used for parachains, so we can safely do this kind of thing.

Reply:
nit: this index is available in the same function above `let new_session_index...`
Reading it from storage again is unnecessary.